transport/pool_integration_test.go

```diff
@@ -50,77 +81,52 @@ func TestNodeConnPoolIntegration(t *testing.T) {
 	}
 }
 
-//func TestCloseConnPoolIntegration(t *testing.T) {
-//	cp := NewConnPool(TestHost+":19042", ConnConfig{})
-//	if cp == nil {
-//		t.Fatal("Couldn't start first connection.")
-//	}
-//
-//	time.Sleep(2 * time.Second)
-//
-//	go func() {
-//		for idx := range cp.conns {
-//			if conn := cp.loadConn(uint16(idx)); conn != nil {
-//				conn.Close()
-//			}
-//		}
-//	}()
-//
-//	cp.Close()
-//
-//	time.Sleep(10 * time.Millisecond)
-//
-//	for idx := range cp.conns {
-//		if conn := cp.loadConn(uint16(idx)); conn != nil {
-//			t.Fatal("Not all connections have been closed.")
-//		}
-//	}
-//}
-//
-//func TestConnPoolConnIntegration(t *testing.T) {
-//	cp := NewConnPool(TestHost+":19042", ConnConfig{})
-//	if cp == nil {
-//		t.Fatal("Couldn't start first connection.")
-//	}
-//
-//	if cp.NextConn() == nil {
-//		t.Fatal("NextConn returns nil even though initConnPool() has been successful.")
-//	}
-//
-//	time.Sleep(2 * time.Second)
-//
-//	murmurToken := MurmurToken([]byte(""))
-//
-//	if conn := cp.Conn(murmurToken); conn == nil || conn.Shard() != 0 {
-//		t.Fatal("Invalid return of Conn.")
-//	}
-//
-//	murmurToken = MurmurToken([]byte("0")) // Very big number approx. 3 * 10^18.
-//
-//	if conn := cp.Conn(murmurToken); conn == nil {
-//		t.Fatal("Invalid return of Conn.")
-//	}
-//
-//	go func() {
-//		seenNil := false
-//		ticker := time.NewTicker(1 * time.Second)
-//		for {
-//			select {
-//			case _ = <-ticker.C:
-//				return
-//			default:
-//				conn := cp.RandConn()
-//				if conn == nil {
-//					seenNil = true
-//				} else if seenNil == true {
-//					t.Fatal("Inconsistent RandConn return.")
-//				}
-//			}
-//		}
-//	}()
-//
-//	cp.Close()
-//
-//	// Waiting for go func().
-//	time.Sleep(1 * time.Second)
-//}
+func TestConnPoolConnIntegration(t *testing.T) {
```

Comment:
Another strange test, what it does?

It would be better to add some token based checks to the 1st test case.

Reply:
I have tried to test here few things:

1. As you have mentioned token shard based checks 
2. The go func() for 100 milliseconds tries to get RandConn while p.Close() is called in the test function. With this test I wanted to make sure that even though successive connections are being closed RandConn is consistent with its returns i.e. there is no case of RandConn returning working connection, next RandConn will return nil and after that RandConn would return another working connection. That would hint that something is wrong with either closing connections or pool refiller still refilling. Maybe it's overthinked but that was my intention.